### PR TITLE
Fix issue #11615  handle unicode encode error when print stack trace

### PR DESCRIPTION
**Pull request by takuya1981**

### CODE REVIEW

**Comment by takuya1981** [comment]
This PR fixed issue #11615

File: IPython/core/interactiveshell.py [hunk 1]
```diff
@@ -2068,13 +2068,17 @@ def showtraceback(self, exc_tuple=None, filename=None, tb_offset=None,
...
 
         Subclasses may override this method to put the traceback on a different
         place, like a side channel.
         """
-        print(self.InteractiveTB.stb2text(stb))
+        val = self.InteractiveTB.stb2text(stb)
+        try:
+            print(val)
+        except UnicodeEncodeError:
+            print(val.encode("utf-8", "backslashreplace").decode())
```
Comment (MrMino) [inline IPython/core/interactiveshell.py:2081]: What are the benefits of having `try...except` here rather than just `print(val.encode(...).decode()`?

Comment (Carreau) [reply IPython/core/interactiveshell.py:2081]: I believe this is question of efficiency. In most case `print()` will work, so there is no need to do the expensive `encode()`, then `.decode()` dance, that would in the end often be no-op.

**Comment by Carreau** [comment]
I think that it's harmless enough that we can get that in. 
Thanks.

**Pull request merged**